python/ray/experimental/state.py

```diff
@@ -543,24 +543,27 @@ def dump_catapult_trace(self,
                 should be included in the trace.
         """
         workers = self.workers()
+
+        task_info = task_info.copy()
```

Comment:
I don't think you need the copy; the below dict comprehension should already generate a new dict.